python/shotgun_desktop/startup.py

```diff
@@ -478,237 +334,118 @@ def __extract_command_line_argument(arg_name):
-                    project_id = pc["project"]["id"]
-                else:
-                    logger.info("Reusing the site config without a project.")
-
-            # Create the directory
-            if not os.path.exists(default_site_config):
-                os.makedirs(default_site_config)
-
-            # Setup the command to create the config
-            if sys.platform == "darwin":
-                path_param = "config_path_mac"
-            elif sys.platform == "win32":
-                path_param = "config_path_win"
-            elif sys.platform.startswith("linux"):
-                path_param = "config_path_linux"
-
-            # allow the config uri to be overridden for testing
-            config_uri = os.environ.get("SGTK_SITE_CONFIG_DEBUG_LOCATION", "tk-config-site")
-
-            params = {
-                "auto_path": True,
-                "config_uri": config_uri,
-                "project_folder_name": "site",
-                "project_id": project_id,
-                path_param: default_site_config,
-            }
-            setup_project = sgtk.get_command("setup_project")
-            setup_project.set_logger(logger)
-
-            try:
-                setup_project.execute(params)
-            except Exception, error:
-                logger.exception(error)
-                if "CRUD ERROR" in error.message:
-                    raise UpdatePermissionsError()
-                else:
-                    raise
-
-            # and now try to load up sgtk through the config again
-            sgtk = __get_initialized_sgtk(default_site_config, app_bootstrap)
-            tk = sgtk.sgtk_from_path(default_site_config)
-
-            # now localize the core to the config
-            splash.set_message("Localizing core...")
-            localize = tk.get_command("localize")
-            localize.set_logger(logger)
-            localize.execute({})
-
-            # Get back the pipeline configuration, this is expected to be initialized further down.
-            _, pc = shotgun_desktop.paths.get_default_site_config_root(connection)
-        except Exception:
-            # Something went wrong. Wipe the default site config if we can and
-            # rethrow
-            if not config_folder_exists_at_startup:
-                logger.error(
-                    "Something went wrong during Toolkit's activation, wiping configuration."
-                )
-                if os.path.exists(default_site_config):
-                    shutil.rmtree(default_site_config)
-            raise
-    else:
-        tk = sgtk.sgtk_from_path(default_site_config)
 
-    # If the pipeline configuration found in Shotgun doesn't match what we have locally, we have a
-    # problem.
-    if pc["id"] != tk.pipeline_configuration.get_shotgun_id():
-        raise InvalidPipelineConfiguration(pc, tk.pipeline_configuration)
+def __bootstrap_toolkit_into_legacy_config(app, splash, user, pc):
```

Comment:
not sure about the use of the word 'legacy' here - maybe it's nitpicky, but it's not legacy just yet. 

This method only seems to be called from `__toolkit_classic_bootstrap` - maybe merge it into that method?